### PR TITLE
Remove unused class_identifier parameter from Java object factory

### PR DESCRIPTION
**Pull request by antlechner**
- [x] Each commit message has a non-empty body, explaining why the change was made.
- n/a Methods or procedures I have added are documented, following the guidelines provided in CODING_STANDARD.md.
- n/a The feature or user visible behaviour I have added or modified has been documented in the User Guide in doc/cprover-manual/
- n/a Regression or unit tests are included, or existing tests cover the modified code (in this case I have detailed which ones those are in the commit message).
- n/a My commit message includes data points confirming performance improvements (if claimed).
- n/a My PR is restricted to a single feature or bugfix.
- n/a White-space or formatting changes outside the feature-related changed lines are in commits of their own.

### CODE REVIEW

File: jbmc/src/java_bytecode/java_object_factory.cpp [hunk 1]
```diff
@@ -826,9 +826,10 @@ void java_object_factoryt::gen_nondet_struct_init(
     // This code mirrors the `remove_java_new` pass:
     auto initial_object = zero_initializer(expr.type(), source_locationt(), ns);
     CHECK_RETURN(initial_object.has_value());
-    const irep_idt qualified_clsid = "java::" + id2string(struct_tag);
     set_class_identifier(
-      to_struct_expr(*initial_object), ns, struct_tag_typet(qualified_clsid));
+      to_struct_expr(*initial_object),
+      ns,
```
Comment (thomasspriggs) [inline jbmc/src/java_bytecode/java_object_factory.cpp:831]: :bulb: `set_class_identifier` doesn't actually use `ns`. So this parameter could probably be removed, however that is probably a change for another PR.

**Review by thomasspriggs — APPROVED** [review]
Neat.

File: jbmc/src/java_bytecode/java_object_factory.cpp (hunk 1 above)
Comment (antlechner) [reply jbmc/src/java_bytecode/java_object_factory.cpp:831]: It has a call to `ns.follow`. I still find it a bit strange that `follow` is defined only for namespaces and not for symbol tables, but that would definitely be a change for another (much bigger) PR. :slightly_smiling_face:

**Review by allredj — COMMENTED** [review]
:heavy_check_mark:
Passed Diffblue compatibility checks (cbmc commit: b97b80aab54d9b67e55e3a6e1f4f5b8b1466e5a9).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/107679876

File: jbmc/src/java_bytecode/java_object_factory.cpp (hunk 1 above)
Comment (thomasspriggs) [reply jbmc/src/java_bytecode/java_object_factory.cpp:831]: Ah, my mistake.

**Review by smowton — APPROVED** [review]

**Review by peterschrammel — APPROVED** [review]

**Pull request merged**